packages/react-core/src/demos/examples/TextInputGroup/AutoCompleteSearch.tsx

```diff
@@ -67,6 +68,20 @@ export const AutoCompleteSearch: React.FunctionComponent = () => {
       return;
     }
 
+    /** The hint is set whenever there is only one autocomplete option left. */
+    if (filteredMenuItems.length === 1) {
+      const hint = filteredMenuItems[0].props.children;
+      if (!inputValue.indexOf(hint)) {
+        // the match was found in a place other than the start, so typeahead wouldn't work right
+        setHint('');
+      } else {
+        // use the input for the first part, otherwise case difference could make things look wrong
+        setHint(inputValue + hint.substr(inputValue.length));
+      }
```

Comment:
Should the `inputValue` and `hint` for the `indexOf` check here be reversed? It looks like the `if` condition will always return false, which might explain the behavior I'm seeing.